src/layout/layout.rs

```diff
@@ -453,6 +453,8 @@ impl Layout {
     pub const fn overlap(mut self, overlap: bool) -> Self {
         if overlap {
             self.spacing = None;
+        } else if self.spacing.is_none() {
+            self.spacing = Some(0)
```

Comment:
This makes `foo.spacing(1).overlap(1)` different to `foo.overlap(1).spacing(1)`. Mostly our builder methods should not have order dependent methods like this.

Reply:
I agree. The only way around this though is by making a breaking change to `spacing`. I'm open to suggestions on what to do here.

Reply:
If these are truly not able to be set together, then they likely deserve to be on an enum with mutually exclusive variants. My initial question on this was on the discord thread sort of had that idea in mind (setting spacing to negative values). If we took that approach, then spacing could be migated to accept `Into<Spacing>` => `enum Spacing { Spaced(u16), Overlap(u16) }`, which would give space for documentation. Alternatively spacing could just become i16 (or i32 if you want to allow layouts that have spacers / overlaps that are > 32767). This would be technically breaking, but not in a way that generally matters to anyone.

Reply:
I think making it `i16` or `i32` would work. It'd break apps that use `spacing` like we do in constraint explorer:

https://github.com/ratatui/ratatui/blob/edcdc8a8147a2f450d2c871b19da6d6383fd5497/examples/constraint-explorer.rs#L47

But maybe that's fine?

Reply:
Removed `overlap` method in https://github.com/ratatui/ratatui/pull/1398/commits/57dbafe8c7fd42d1901cbc41810b3c609f578ae0